storagedriver/azure/azure.go

```diff
@@ -272,6 +361,24 @@ func (d *Driver) Delete(path string) error {
 	return nil
 }
 
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *Driver) URLFor(path string, options map[string]interface{}) (string, error) {
```

Comment:
Yep, this is creates a publicly accessible URL for given period of time. It is indeed a signed URL. http://azure.microsoft.com/en-us/documentation/articles/storage-dotnet-shared-access-signature-part-1/